java8/src/main/java/com/dslplatform/json/processor/ConverterTemplate.java

```diff
@@ -227,7 +265,11 @@ private void asFormatConverter(final StructInfo si, final String name, final Str
 					if (attr.isArray) {
 						type = typeForGeneric(((ArrayType) attr.type).getComponentType(), attr.typeVariablesIndex);
 					} else {
-						type = typeForGeneric(attr.type, attr.typeVariablesIndex);
+					    if (si.isParameterized) {

```

Comment:
Can you pretty please fix your alignments?

Reply:
Yes, of course.